aws/provider.go

```diff
@@ -525,6 +525,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_lambda_event_source_mapping":                  resourceAwsLambdaEventSourceMapping(),
 			"aws_lambda_alias":                                 resourceAwsLambdaAlias(),
 			"aws_lambda_permission":                            resourceAwsLambdaPermission(),
+			"aws_lambda_layer":                                 resourceAwsLambdaLayer(),
```

Comment:
Hot take 🔥: I think we should follow AWS CloudFormation/CLI on this one and add `_version` in the Terraform resource name  to explicitly signify this is managing just a single version of a Lambda Layer. I could see it being confusing ("why do I need to delete my Lambda Layer?") with the Lambda API's lack of update functionality for these.

Unfortunately this type of change requires updates to each of the filenames and function names.

Reply:
makes sense - I'll do that